File: TowersOfHanoi.java
import java.util.Stack; // Not in LANG!!!

public class TowersOfHanoi {

public static void main(String[] args){
        Stack<Integer> s = new Stack[args[0]];
}

public void solver(int disks, Stack origin, Stack destination, Stack via){
        if (disks >= 1) {
                solver(disks - 1, origin, via, destination);
                move(disks, origin, destination);
                solver(disks - 1, via, destination, origin);
        }
}

public void move(int disk, Stack origin, Stack destination){
        destination.push(origin.pop());
}
}

File: TowersOfHanoi.py

def TowersOfHanoi(n, s, d, v):
    if n >= 1:
        TowersOfHanoi(n - 1, s, v, d)
        move(s, d)
        TowersOfHanoi(n - 1, v, d, s)


def move(s, d):
    print "Moving {} to {}".format(s.pop(0), d)


s = [x for x in range(1, 9)]
print s
d = []
v = []
TowersOfHanoi(5, s, d, v)
